ui/revenuecatui/src/test/kotlin/com/revenuecat/purchases/ui/revenuecatui/utils/FontFamilyXMLParserTest.kt

```diff
@@ -0,0 +1,238 @@
+            </font-family>"""
+        
+        val parser = TestXmlResourceParser(xmlContent, mapOf("@font/regular" to 789))
+        val result = FontFamilyXMLParser.parseXmlData(parser)
+        
+        assertThat(result).hasSize(1)
+        assertThat(result[0].first).isEqualTo(789)
+        assertThat(result[0].second).isEqualTo(400) // default weight
+        assertThat(result[0].third).isEqualTo(FontStyle.Normal) // default style
+    }
+
+    @Test
+    fun `parseXmlData returns empty list when font resource ID is invalid`() {
+        val xmlContent = """<?xml version="1.0" encoding="utf-8"?>
+            <font-family xmlns:app="http://schemas.android.com/apk/res-auto">
+                <font app:fontStyle="normal" app:fontWeight="400" />
+            </font-family>"""
+        
+        val parser = TestXmlResourceParser(xmlContent) // no resource mapping
+        val result = FontFamilyXMLParser.parseXmlData(parser)
+        
+        assertThat(result).isEmpty()
+    }
+
+    @Test
+    fun `parseXmlData falls back from app namespace to android namespace`() {
+        val xmlContent = """<?xml version="1.0" encoding="utf-8"?>
+            <font-family 
+                xmlns:app="http://schemas.android.com/apk/res-auto"
+                xmlns:android="http://schemas.android.com/apk/res/android">
+                <font android:fontStyle="normal" android:fontWeight="400" android:font="@font/regular" />
+            </font-family>"""
+        
+        val parser = TestXmlResourceParser(xmlContent, mapOf("@font/regular" to 999))
+        val result = FontFamilyXMLParser.parseXmlData(parser)
+        
+        assertThat(result).hasSize(1)
+        assertThat(result[0].first).isEqualTo(999)
+        assertThat(result[0].second).isEqualTo(400)
+        assertThat(result[0].third).isEqualTo(FontStyle.Normal)
+    }
+
+    @Test
+    fun `parse returns null for empty XML`() {
+        val xmlContent = """<?xml version="1.0" encoding="utf-8"?>
+            <font-family xmlns:android="http://schemas.android.com/apk/res/android">
+            </font-family>"""
+        
+        val parser = TestXmlResourceParser(xmlContent)
+        val result = FontFamilyXMLParser.parse(parser)
+        
+        assertThat(result).isNull()
+    }
+
+    @Test
+    fun `parse returns FontFamily when fonts are parsed successfully`() {
+        val xmlContent = """<?xml version="1.0" encoding="utf-8"?>
+            <font-family xmlns:app="http://schemas.android.com/apk/res-auto">
+                <font app:fontStyle="normal" app:fontWeight="400" app:font="@font/regular" />
+            </font-family>"""
+        
+        val parser = TestXmlResourceParser(xmlContent, mapOf("@font/regular" to 123))
+        val result = FontFamilyXMLParser.parse(parser)
+        
+        assertThat(result).isNotNull()
+    }
+
+    // Test implementation of XmlResourceParser that uses real XML parsing
+    private class TestXmlResourceParser(
+        xmlContent: String,
```

Comment:
If we add this annotation, Android Studio will treat every string passed-in as actual XML and do syntax highlighting and allows you to format it and stuff:

```suggestion
        @Language("xml") xmlContent: String,
```

(A `// language=xml` comment above every field works too, but that needs to be repeated everywhere.)

You can do Option+Enter and "Edit XML Fragment" to format it.
<img width="499" height="297" alt="image" src="https://github.com/user-attachments/assets/367299bb-9bf3-48c9-b33b-84bd1ac8aada" />

And it will underline syntax errors:
<img width="722" height="161" alt="image" src="https://github.com/user-attachments/assets/aaf64383-12ba-49eb-8a73-904f2b59bf5d" />

Reply:
Oh nice! TIL!